service/Abstractions/Models/MemoryFilter.cs

```diff
@@ -17,18 +24,75 @@ public MemoryFilter ByTag(string name, string value)
         return this;
     }
 
+    public MemoryFilter ByNotTag(string name, string value)
+    {
+        this._notTags.Add(name, value);
+        return this;
+    }
+
     public MemoryFilter ByDocument(string docId)
     {
         this.Add(Constants.ReservedDocumentIdTag, docId);
         return this;
     }
 
+    /// <summary>
+    /// Get all the filters you need to put into the query
+    /// </summary>
+    /// <returns></returns>
     public IEnumerable<KeyValuePair<string, string?>> GetFilters()
     {
         return this.ToKeyValueList();
     }
+
+    /// <summary>
+    /// Gets all the filters that needs to be put as not into  the query
+    /// </summary>
+    /// <returns></returns>
+    public IEnumerable<KeyValuePair<string, string?>> GetNotFilters()
+    {
+        return this._notTags.ToKeyValueList();
+    }
+
+    /// <summary>
+    /// Get a composition of all filters, And and Not.
+    /// </summary>
+    /// <returns></returns>
+    public IEnumerable<BaseFilter> GetAllFilters()
```

Comment:
What about changing the implementation of `GetFilters` rather than introducing this new method?

QdrantMemory is still using `GetFilters` in `GetListAsync`, missing the NOt-filters.

Reply:
You are right, actually I misses that implementation of qdrant, I'll fix and then the original GetFilters is not called anymore, so we can actually rename this in GetFilters

[Done]